### PR TITLE
fix: psmCharter.addInstance interface shape correction

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
## Description

Correct the interface shape declaration for `psmCharter.addInstance()`.  The last two parameters are optional.

### Security Considerations

None

### Documentation Considerations

None

### Testing Considerations

Exercised by running `make scenario2-setup scenario2-run-chain-psm`.

### CODE REVIEW

File: packages/inter-protocol/src/psm/psmCharter.js [hunk 1]
```diff
@@ -96,12 +96,9 @@ export const start = async zcf => {
...
 
   const psmCharterCreatorI = M.interface('PSM Charter creatorFacet', {
-    addInstance: M.call(
-      InstanceHandleShape,
-      M.any(),
-      BrandShape,
-      BrandShape,
-    ).returns(),
+    addInstance: M.call(InstanceHandleShape, M.any())
+      .optional(BrandShape, BrandShape)
```
Comment (turadg) [inline packages/inter-protocol/src/psm/psmCharter.js:100]: TIL `.optional`. Where is it documented or implemented?

**Review by turadg — APPROVED** [review]
`.optional` is new to me but if it works for you locally then I believe it. 

I added the `bypass:integration` label. Since the test didn't catch the bug going it it shouldn't be affected by it being fixed ;-)

**Review by dckc — APPROVED** [review]

File: packages/inter-protocol/src/psm/psmCharter.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/src/psm/psmCharter.js:100]: There's [very little documentation](https://github.com/Agoric/agoric-sdk/blob/c10c36d7ccf6c85239c1dbcec9534d43b20ad00a/packages/store/src/types.js).  I dunno where the implementations are. It's tested in [test-heap-classes](https://github.com/Agoric/agoric-sdk/blob/c10c36d7ccf6c85239c1dbcec9534d43b20ad00a/packages/store/test/test-heap-classes.js#L14).

Comment (turadg) [reply packages/inter-protocol/src/psm/psmCharter.js:100]: Yeah, that documentation doesn't mention `optional`. I've filed https://github.com/Agoric/agoric-sdk/issues/6206 cc @erights

Comment (erights) [reply packages/inter-protocol/src/psm/psmCharter.js:100]: From https://github.com/Agoric/agoric-sdk/issues/6206#issuecomment-1247546215 :

FWIW, this is implemented at https://github.com/Agoric/agoric-sdk/blob/9b786284b4ff759dd050c701829df5fd0cf41749/packages/store/src/patterns/patternMatchers.js#L1537-L1543 and used several times in https://github.com/Agoric/agoric-sdk/blob/master/packages/ERTP/src/typeGuards.js

I'm actually surprised I didn't write any comments explaining this at all. I certainly agree it is needed.

Also `rest`

**Pull request merged**